src/domUpdates.js

```diff
@@ -28,19 +30,68 @@ let domUpdates = {
           `
       )
     })
-    // getFavorites();
   },
-  cardButtonConditionals(event) {
-    console.log(event.target)
-    if (event.target.classList.contains('favorite')) {
-      favoriteCard(event);
-    } else if (event.target.classList.contains('card-picture')) {
-      displayDirections(event);
-    } else if (event.target.classList.contains('home')) {
+  viewFavorites(cardArea, user, favButton, currentFavs,) {
+    if (!user.favoriteRecipes.length) {
+      favButton.innerHTML = 'You have no favorites!';
+      return
+    } else {
+      favButton.innerHTML = 'Refresh Favorites'
+      cardArea.innerHTML = '';
+      this.populateCards(user.favoriteRecipes, cardArea, currentFavs)
+    }
+  },
+  displayFavorite(event, favButton) {
+    if (!event.target.classList.contains('favorite-active')) {
+      event.target.classList.add('favorite-active');
       favButton.innerHTML = 'View Favorites';
-      populateCards(cookbook.recipes);
+      return  true
+    } else if (event.target.classList.contains('favorite-active')) {
+      event.target.classList.remove('favorite-active');
+      return false
     }
-  }
+  },
+  displayDirections(cardArea, recipeObject, costInDollars) {
+    cardArea.classList.add('all');
+    cardArea.innerHTML = `
+      <h3>${recipeObject.name}</h3>
+        <p class='all-recipe-info'>
+        <strong>It will cost: </strong><span class='cost recipe-info'>
+        $${costInDollars}</span><br><br>
+        <strong>You will need: </strong><span class='ingredients recipe-info'></span>
+        <strong>Instructions: </strong><ol><span class='instructions recipe-info'>
+        </span></ol>
+      </p>`;
+    let ingredientsSpan = document.querySelector('.ingredients');
+    let instructionsSpan = document.querySelector('.instructions');
+    recipeObject.ingredients.forEach(ingredient => {
+      ingredientsSpan.insertAdjacentHTML('afterbegin', `<ul><li>
+      ${ingredient.quantity.amount.toFixed(2)} ${ingredient.quantity.unit}
+      ${ingredient.name}</li></ul>
+      `)
+    })
+    recipeObject.instructions.forEach(instruction => {
+      instructionsSpan.insertAdjacentHTML('beforebegin', `<li>
+      ${instruction.instruction}</li>
+      `)
+    })
+  },
+  updateSearch() {
+    let recipeNames = document.querySelectorAll('.recipe-name');
+    recipeNames.forEach(recipe =>{
+      let lowerCaseRecipe = recipe.innerText.toLowerCase()
+      if (!lowerCaseRecipe.includes(searchInput.value)) {
+        document.getElementById(recipe.id).classList.add("hidden")
+      } else {
+        document.getElementById(recipe.id).classList.remove("hidden");
```

Comment:
This is currently not working, but just needs to be slightly adjusted in order to do so.